section_faiss_vector_index.go

```diff
@@ -271,85 +271,64 @@ func (v *vectorIndexOpaque) mergeAndWriteVectorIndexes(fieldID int, sbs []*Segme
 
 	defer freeReconstructedIndexes(vecIndexes)
 
-	// todo: perhaps making this threshold a config value would be better?
-	if len(vecToDocID) > 10000 {
-		// merging of more complex index types (for eg ivf family) with reconstruction
-		// method.
-		var indexData []float32
-		for i := 0; i < len(vecIndexes); i++ {
-			if isClosed(closeCh) {
-				return fmt.Errorf("merging of vector sections aborted")
-			}
-			// todo: parallelize reconstruction
-			recons, err := vecIndexes[i].ReconstructBatch(int64(len(indexes[i].vecIds)), indexes[i].vecIds)
-			if err != nil {
-				return err
-			}
-			indexData = append(indexData, recons...)
-		}
-
-		// safe to assume that all the indexes are of the same config values, given
-		// that they are extracted from the field mapping info.
-		dims := vecIndexes[0].D()
-		metric := vecIndexes[0].MetricType()
-		finalVecIDs := maps.Keys(vecToDocID)
-
-		// todo: perhaps the index type can be chosen from a config and tuned accordingly.
-		index, err := faiss.IndexFactory(dims, "IVF100,SQ8", metric)
-		if err != nil {
-			return err
-		}
-
-		defer index.Close()
-
-		// the direct map maintained in the IVF index is essential for the
-		// reconstruction of vectors based on vector IDs in the future merges.
-		// the AddWithIDs API also needs a direct map to be set before using.
-		err = index.SetDirectMap(2)
-		if err != nil {
-			return err
-		}
-
-		// train the vector index, essentially performs k-means clustering to partition
-		// the data space of indexData such that during the search time, we probe
-		// only a subset of vectors -> non-exhaustive search. could be a time
-		// consuming step when the indexData is large.
-		err = index.Train(indexData)
-		if err != nil {
-			return err
+	// merging of more complex index types (for eg ivf family) with reconstruction
+	// method.
+	var indexData []float32
+	for i := 0; i < len(vecIndexes); i++ {
+		if isClosed(closeCh) {
+			return fmt.Errorf("merging of vector sections aborted")
 		}
-
-		index.AddWithIDs(indexData, finalVecIDs)
+		// todo: parallelize reconstruction
+		recons, err := vecIndexes[i].ReconstructBatch(int64(len(indexes[i].vecIds)), indexes[i].vecIds)
 		if err != nil {
 			return err
 		}
+		indexData = append(indexData, recons...)
+	}
 
-		mergedIndexBytes, err := faiss.WriteIndexIntoBuffer(index)
-		if err != nil {
-			return err
-		}
+	// safe to assume that all the indexes are of the same config values, given
+	// that they are extracted from the field mapping info.
+	dims := vecIndexes[0].D()
+	metric := vecIndexes[0].MetricType()
+	finalVecIDs := maps.Keys(vecToDocID)
+
+	var indexType string
+	// todo: perhaps the index type can be chosen from a config and tuned accordingly.
+	switch {
+	case len(vecToDocID) > 10000:
+		indexType = "IVF100,SQ8"
+	default:
+		indexType = "IVF10,Flat"
```

Comment:
Putting in a requirement of at least 10 vectors to survive the merge is not a good idea.